Assets/BossRoom/Scripts/Client/UI/UnityServicesUIHandler.cs

```diff
@@ -49,25 +49,38 @@ void ServiceErrorHandler(UnityServiceErrorMessage error)
 
         void HandleLobbyError(UnityServiceErrorMessage error)
         {
-            var errorMessage = error.Message;
-            switch (((LobbyServiceException)error.OriginalException).Reason)
+            var exception = (LobbyServiceException) error.OriginalException;
```

Comment:
I think that you'd get a miscast exception if OriginalException is something else than LobbyServiceException.

Reply:
Normally it should not happen since we only call HandleLobbyError when error.AffectedService equals UnityServiceErrorMessage.Service.Lobby. If it does, it would mean that the UnityServiceErrorMessage was not properly constructed, right?

Reply:
Well, HandleLobbyError method is presuming that no other UnityServices will ever use UnityServiceErrorMessage or something like that That's a wrong assumption. Exception is the type of OriginalException. Everything else is just a possibility.